src/Bicep.Local.Deploy/Extensibility/LocalExtensibilityHandler.cs

```diff
@@ -20,22 +25,18 @@ private record ProviderKey(
         string Name,
         string Version);
 
-    private Dictionary<ProviderKey, Lazy<Task<LocalExtensibilityProvider>>> RegisteredProviders = new();
+    private Dictionary<ProviderKey, LocalExtensibilityProvider> RegisteredProviders = new();
+    private readonly IModuleDispatcher moduleDispatcher;
+    private readonly Func<Uri, Task<LocalExtensibilityProvider>> providerFactory;
 
-    private LocalExtensibilityHandler()
+    public LocalExtensibilityHandler(IModuleDispatcher moduleDispatcher, Func<Uri, Task<LocalExtensibilityProvider>> providerFactory)
     {
+        this.moduleDispatcher = moduleDispatcher;
+        this.providerFactory = providerFactory;
+        // Built in provider for handling nested deployments
+        RegisteredProviders[new("LocalNested", "0.0.0")] = new AzExtensibilityProvider(this);
```

Comment:
Don't recall if we have a way to load nested deployments provider version, I think it might be good to let users decide which version to use rather than picking a specific version?

Reply:
Bicep doesn't expose the nested deployment API version to end users in the `module` syntax, so it's not possible to pick a different version. This is by design - Bicep should be in control of the behavior; if we need to opt-in to new behavior, then we will do so in a new Bicep release.